codegen/masm/tests/test_basic_arithmetic.rs

```diff
@@ -0,0 +1,227 @@
+use air_codegen_masm::{code_gen, constants};
+use assembly::Assembler;
+use ir::AirIR;
+use miden_diagnostics::{
+    term::termcolor::ColorChoice, CodeMap, DefaultEmitter, DiagnosticsHandler,
+};
+use parser::{ast::Source, Parser};
+use processor::{AdviceInputs, Kernel, MemAdviceProvider, Process, StackInputs};
+use std::sync::Arc;
+
+pub fn parse(source: &str) -> Source {
+    let codemap = Arc::new(CodeMap::new());
+    let emitter = Arc::new(DefaultEmitter::new(ColorChoice::Auto));
+    let diagnostics = DiagnosticsHandler::new(Default::default(), codemap.clone(), emitter);
+    let parser = Parser::new((), codemap);
+
+    parser
+        .parse_string::<Source, _, _>(&diagnostics, source)
+        .expect("parsing failed")
+}
+
+static ARITH_AIR: &str = "
+def Add
+
+trace_columns:
+    main: [a, b]
+
+public_inputs:
+    stack_inputs: [16]
+
+boundary_constraints:
+    enf a.first = 0
```

Comment:
I'd probably add a constraint against the last step as well, and maybe something which involves public inputs.

Reply:
Agreed. Note that this is not the complete backend, I'm trying to open a PR to get input early on. This constraint is currently *not* evaluated, since the ATM the backend is only generating the MASM for the transition constraints. Currently this is only here to satisfy the Air parser, once the boundary constraint is implemented I will add more complex tests.